hca/upload/api_client.py

```diff
@@ -8,8 +10,11 @@ class ApiClient:
     def __init__(self, deployment_stage):
         self.api_url_base = ApiClient.UPLOAD_SERVICE_API_URL_TEMPLATE.format(deployment_stage=deployment_stage)
 
-    def list_area(self, area_uuid):
-        url = "{api_url_base}/area/{uuid}".format(api_url_base=self.api_url_base, uuid=area_uuid)
-        response = requests.get(url)
-        assert response.ok
-        return response.json()['files']
+    def files_info(self, area_uuid, file_list):
+        url = "{api_url_base}/area/{uuid}/files_info".format(api_url_base=self.api_url_base, uuid=area_uuid)
+        response = requests.put(url, data=(json.dumps(file_list)))
```

Comment:
I assume this is because GET has a lower payload size limit than PUT?  Kind of unfortunate... :(

Reply:
I was originally using GET, but I discovered that the AWS API Gateway throws away your payload for a GET, the HTTP spec says that a server response to a GET should not vary based on the payload, and as Andrey pointed out, if it does that breaks caching.  So in this case we have to abandon RESTfullness and just call it an RPC call :)